mrao2/newtowndogs
# Appointment Booking and Appointment Review

This pull request merges in my latest updates to the appointment request page, the form submission notification page, the appointment requests review page, and the mobile version of the responsive navigation bar. 

*****PRIOR TO** merging my changes into your main, or checking out my changes to approve the request, please run the following code to install [Font Awesome](https://fontawesome.com/). Font Awesome allows you to add icons to a website.

npm i --save @fortawesome/fontawesome-svg-core

npm install --save @fortawesome/free-solid-svg-icons

npm install --save @fortawesome/react-fontawesome

THEN, you need to drop and replace your appointments table. The new SQL is in the SQL folder (same place as the old SQL).

***

To Test:

1. the mobile version of the responsive navigation bar and Font Awesome:
- Resize your browser to see if the navigation bar changes correctly. 

<img width="498" alt="cropped" src="https://github.com/mrao2/newtowndogs/assets/76797183/34c75548-cc24-4013-a2e4-9268f41a43ab">

2. the appointment request form:
- Select the Book Sitter link from the navigation bar.
- Using the form create two appointments: a consultation appointment and a returning customer appointment. Use the checkbox to toggle between the two.

<img width="498" alt="Screen Shot 2023-08-10 at 6 21 09 PM" src="https://github.com/mrao2/newtowndogs/assets/76797183/f209f584-b716-451a-8bc7-b6b40e18e5e9">

- After creating each appointment, you should be redirected to an appointment request submission confirmation page.

<img width="501" alt="Screen Shot 2023-08-10 at 6 26 38 PM" src="https://github.com/mrao2/newtowndogs/assets/76797183/214da78f-f4d3-4185-9b46-8eccdca72116">

3. the appointment form data has correctly populated the table in the MySQL database:
- Note that there will be empty fields, and this is okay.
- Check that you have one consultation appointment and one return customer appointment.

<img width="950" alt="Screen Shot 2023-08-14 at 5 24 14 PM" src="https://github.com/mrao2/newtowndogs/assets/76797183/306fec82-94c0-4cbd-bc5c-c5b6972c483c">

4. the appointment requests display on the Appointment Requests page:
- The appointments will be numbered with their appointment_id.
- They will resize responsively if the display size changes.
- They have correct data, and that null fields display "---".

<img width="1082" alt="Screen Shot 2023-08-10 at 6 17 55 PM" src="https://github.com/mrao2/newtowndogs/assets/76797183/686d580b-c2ed-42bd-85d0-81ec1d041e18">

5. the appointment request review features (confirm, reject, and delete):

***Please note that in future updates I plan to connect the confirm and reject buttons to the database, which will update the currently unused fields in the table. This has not been done yet.***

- Clicking the confirm button will open your computer email client with the email address and subject line populated correctly.
- Clicking the reject button will open your computer email client with the email address and subject line populated correctly.
- Clicking the trash can button deletes the appointment row from the database and refreshes the page.

<img width="487" alt="Screen Shot 2023-08-14 at 5 36 50 PM" src="https://github.com/mrao2/newtowndogs/assets/76797183/6b86018c-b92a-4e71-8656-674c9984649b">
<img width="494" alt="Screen Shot 2023-08-14 at 5 38 56 PM" src="https://github.com/mrao2/newtowndogs/assets/76797183/c57184ff-7222-44d4-a23e-21ad25ecb369">
